v2/server/controllers/user.controller.js

```diff
@@ -0,0 +1,63 @@
+import users from '../models/user.model';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import queries from '../config/queries';
+import executeQuery from "../config/connectDb";
+import userValidations from '../helpers/user.validation';
+
+dotenv.config();
+
+class UserController {
+    static async signup(req, res) {
+        try{
+        const { error } = userValidations.validateSignup(req.body);
+        if (error){
+            return res.status(400).send({ status: 400, error: error.details[0].message });
+        }
+        const { firstName, lastName, email, address, gender, jobRole, department } = req.body;
+        const isUserExist = await executeQuery(queries[0].isUserExist, [email]);
+        if (isUserExist[0]) {
+            return res.status(409).send({
+                status: 409,
+                message: "user already exist"
```

Comment:
if we are checking the email it is best to return a response like `email already in use`